tools/ipa/rulesets/functions/eachResourceHasGetMethod.js

```diff
@@ -0,0 +1,95 @@
+const ERROR_MESSAGE = "APIs must provide a get method for resources."
+
+export default (input, _, context) => {
+  if (isChild(input) || isCustomMethod(input)) {
+    return
+  }
+
+  const oas = context.documentInventory.resolved
+
+  const resourcePaths = getResourcePaths(input, Object.keys(oas.paths))
+
+  if (isSingletonResource(resourcePaths)) {
+    // Singleton resource, may have custom methods
+    if (!hasGetMethod(oas.paths[resourcePaths[0]])) {
+      return [
+        {
+          message: ERROR_MESSAGE
+        }
+      ]
+    }
+  } else if (isNormalResource(resourcePaths)) {
+    // Normal resource, may have custom methods
+    if (!hasGetMethod(oas.paths[resourcePaths[1]])) {
+      return [
+        {
+          message: ERROR_MESSAGE
+        }
+      ]
+    }
+  }
+}
+
+function isChild(path) {
+  return path.endsWith("}")
+}
+
+function isCustomMethod(path) {
+  return path.includes(":")
+}
```

Comment:
Can these be defined in a separate file and imported here, I may answer one of @drinkbird questions

Reply:
Definitely, I expect we'll have a bunch of common helper functions we can import for different custom functions